src/napari_spatialdata/_viewer.py

```diff
@@ -0,0 +1,187 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+from anndata import AnnData
+from loguru import logger
+from multiscale_spatial_image import MultiscaleSpatialImage
+from napari import Viewer
+from napari.layers import Labels, Points, Shapes
+from napari.utils.notifications import show_info
+
+from napari_spatialdata.utils._utils import _get_transform, _swap_coordinates, _transform_to_rgb
+
+if TYPE_CHECKING:
+    from napari.layers import Layer
+    from spatialdata import SpatialData
+
+
+class SpatialDataViewer:
+    def __init__(self, viewer: Viewer) -> None:
+        self.viewer = viewer
+        self.viewer.bind_key("Shift-L", self._inherit_metadata)
+
+    def _inherit_metadata(self, viewer: Viewer) -> None:
+        layers = list(viewer.layers.selection)
+        self.inherit_metadata(layers)
+
+    def inherit_metadata(self, layers: list[Layer]) -> None:
```

Comment:
I think this approach is a clean and safe one, so I would merge the PR now. But I wonder if we can avoid to have the user having to know that this shortcut exists, to make the user experience more intuitive. We could explore this in a follow up PR so that we unlock this one.

One idea that we could consider for instance is to take advantage that the user will have to press `Shift+E` to save the annotations to disk. So maybe we could go back to place the metadata in every layer that is added, but ignore the metadata in those layers. 

So for instance, even if an external plugin is adding layers which then will be populated with some metadata, in particualar `_active_in_cs`, practically no behavior will be induced by the presence or absence of those metadata. But when the user presses `Shift+E` to save a layer, then those metadata will become "active", and we will consider them.

Wdyt?

Reply:
I am unsure whether I understand you correctly, but I think this would be quite challenging from based on my current understanding of your proposal. Differentiating between layers that are already backed by a `SpatialData` storage and those that are not in that case would require an additional check before applying a certain behavior that is currently there. It also forces the user to practically save first in order to make particular behaviour available.

That being said, in the long run I think we can avoid this completely with true multicanvas view where we have the `SpatialData` layers on one canvas and anything else on the other.

Reply:
Ok, so then let's keep the currently implemented behavior and see later how to remodel it with multicanvas.

Reply:
I agree it would be nice to do this automatically. However, I think it's difficult to do so at the moment, so I like the approach of having a manual approach backed by a public method that can be used whenever we figure out how to do it automatically.